qa-pipelines/tasks/lib/azure-aks.sh

```diff
@@ -49,16 +45,30 @@ azure_check_lbs_ready_in_namespace() {
     # checks that all LoadBalancer type services in namespace have an ingress IP
     local namespace=$1
     local lb_info lb_count lb_ready_count
-    lb_info=$(kubectl get svc -n $namespace -o json | jq -c '[.items[] | select(.spec.type == "LoadBalancer") | {"svc": .metadata.name, "ip": .status.loadBalancer.ingress[0].ip}]')
+    lb_info=$(
+      kubectl get svc -n $namespace -o json |
+      jq -c '
+        [
+          .items[] | select(.spec.type == "LoadBalancer") | {
+            "svc": .metadata.name,
+            "ip": .status.loadBalancer.ingress[0].ip,
+            "hostname": .status.loadBalancer.ingress[0].hostname
+           }
+        ]'
+    )
     lb_count=$(echo "${lb_info}" | jq length)
-    lb_ready_count=$(echo "${lb_info}" | jq -r '[.[] | select(.ip)] | length')
+    if [[ ${cap_platform} == "eks" ]]; then
+        lb_ready_count=$(echo "${lb_info}" | jq -r '[.[] | select(.hostname)] | length')
+    else
+        lb_ready_count=$(echo "${lb_info}" | jq -r '[.[] | select(.ip)] | length')
+    fi
     [[ ${lb_count} -eq ${lb_ready_count} ]]
 }
 
 azure_wait_for_lbs_in_namespace() {
     local namespace=$1
     local count=0
-    while ! azure_check_lbs_ready_in_namespace $namespace; do
+    while ! azure_check_lbs_ready_in_namespace $namespace $cap_platform; do
```

Comment:
$cap_platform as a parameter isn't being used; the function receiving it gets it because it's set globally, so no need to pass it as a parameter